src/Data/Formatter/XmlDataResponseFormatter.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Data\Formatter;
+
+use DOMDocument;
+use DOMElement;
+use DOMException;
+use DOMText;
+use Psr\Http\Message\ResponseInterface;
+use Yiisoft\Strings\StringHelper;
+use Yiisoft\Yii\Web\Data\DataResponse;
+use Yiisoft\Yii\Web\Data\DataResponseFormatterInterface;
+
+final class XmlDataResponseFormatter implements DataResponseFormatterInterface
+{
+    /**
+     * @var string the Content-Type header for the response
+     */
+    private string $contentType = 'application/xml';
+    /**
+     * @var string the XML version
+     */
+    private string $version = '1.0';
+    /**
+     * @var string the XML encoding.
+     */
+    private string $encoding = 'UTF-8';
+    /**
+     * @var string the name of the root element. If set to false, null or is empty then no root tag should be added.
+     */
+    private string $rootTag = 'response';
+    /**
+     * @var string the name of the elements that represent the array elements with numeric keys.
+     */
+    private string $itemTag = 'item';
+    /**
+     * @var bool whether to interpret objects implementing the [[\Traversable]] interface as arrays.
+     * Defaults to `true`.
+     */
+    private bool $useTraversableAsArray = true;
+    /**
+     * @var bool if object tags should be added
+     */
+    private bool $useObjectTags = true;
+
+    public function format(DataResponse $dataResponse): ResponseInterface
+    {
+        $content = '';
+        $data = $dataResponse->getData();
+        if ($data !== null) {
+            $dom = new DOMDocument($this->version, $this->encoding);
```

Comment:
Please create new `Serializer` in `yiisoft/serializer` and use it instead

Reply:
That is planned: https://github.com/yiisoft/serializer/issues/5